rfc/rfc003-oauth2-authorization.md

```diff
@@ -101,19 +102,20 @@ Each compound service MUST define an `oauth` serviceEndpoint. This endpoint refe
 In order for the client to resolve the authorization server endpoint it MUST look up the `oauth` service endpoint of the compound service.
 A compound service may refer to endpoint services from other DID Documents.
 
-### 4.2. Constructing the JWT
+### 4.2 Constructing the JWT
 
-#### 4.2.1. Header
+#### 4.2.1 Header
 
 * **typ**: MUST be `JWT`
 * **alg**: one of `PS256`, `PS384`, `PS512`, `ES256`, `ES384` or `ES512` \([RFC7518](https://tools.ietf.org/html/rfc7518)\)
 * **kid**: MUST contain the identifier of a key published in the actor's DID document, listed in the `assertionMethod` section.  
 
-#### 4.2.2. Payload
+#### 4.2.2 Payload
 
 * **iss**: The issuer MUST contain the DID of the actor, thus the care organization making the request.
 * **sub**: The subject MUST contain the DID of the custodian. The custodian's DID could be used to find the relevant consent \(together with the actor and subject\).
 * **vcs**: A list of base64 encoded [Nuts Authorization Credentials](rfc014-authorization-credential.md). Optional
+* **purposeOfUse**: A list of desired usages. Corresponds to Bolts. Omitted when empty.   
```

Comment:
If feels like `purposeOfUse` has the same meaning as scopes in the oauth2 spec. Can't we use the same terminology and usage of oauth2 scopes?

Reply:
`scope` has specific requirements. For example, it's specified by the authorization server. So there's not a 1-on-1 match which makes it feel like abusing it.

Reply:
> scope has specific requirements

Yes, it has to be a string and it's case sensitive. But for our use-case that's fine right?


> it's specified by the authorization server
> there's not a 1-on-1 match

Yeah, that feels a little bit weird indeed. But there is a 1-on-1 match right? As a scope will be the name of a bolt in this case.

Reply:
In this case I think the `scope` property would be a good fit for this. The benefit is also that people will understand what it's for without thinking as it's part of the oauth2 spec.

Reply:
Nevermind, it's not a good idea as the `purposeOfUse` has to be there before the actual oauth endpoint (`/n2n/auth/v1/accesstoken`) right?

Reply:
> > scope has specific requirements
> 
> Yes, it has to be a string and it's case sensitive. But for our use-case that's fine right?
> 
> > it's specified by the authorization server
> > there's not a 1-on-1 match
> 
> Yeah, that feels a little bit weird indeed. But there is a 1-on-1 match right? As a scope will be the name of a bolt in this case.

I think purposeOfUse will contain policies specified by the Bolt. So in some cases it may be the BOlt (because it specifies just 1 policy), but it might be more?